packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/log_stream_provider.dart

```diff
@@ -9,10 +11,12 @@ import 'package:intl/intl.dart';
 /// [CloudWatchLogStreamProvider]
 /// {@endtemplate}
 abstract class CloudWatchLogStreamProvider {
-  /// Returns CloudWatch log stream name to use for sending logs to CloudWatch.
-  ///
-  /// It creates the log stream if it does not exists.
-  Future<String> get logStream;
+  /// Returns log stream name from cache. if cache is missing it
+  /// calls [createLogStream] and return the log stream name.
+  FutureOr<String> get logStream;
```

Comment:
I would maybe change the name to `currentLogStream` or similar to distinguish from `createLogStream`. Is this class going to be public?

Reply:
the DefaultCloudWatchLogStreamProvider is not going to be public. should I move it to the `cloudwatch_logger_plugin.dart` and make it private?